src/librustc_typeck/astconv.rs

```diff
@@ -1021,16 +1012,26 @@ impl<'o, 'gcx: 'tcx, 'tcx> dyn AstConv<'gcx, 'tcx>+'o {
             return tcx.types.err;
         }
 
-        // Use a BTreeSet to keep output in a more consistent order.
+        // Use a `BTreeSet` to keep output in a more consistent order.
         let mut associated_types = BTreeSet::default();
 
         for tr in traits::supertraits(tcx, principal) {
             associated_types.extend(tcx.associated_items(tr.def_id())
                 .filter(|item| item.kind == ty::AssociatedKind::Type)
                 .map(|item| item.def_id));
+
+            projection_bounds.extend(tcx.predicates_of(tr.def_id())
```

Comment:
The argument to `extend` warrants its own function (the same code also appears elsewhere)

Reply:
Actually I just forgot to remove it in one place I think heh.